cluster.go

```diff
@@ -620,7 +620,10 @@ func (c *Cluster) Shutdown(ctx context.Context) error {
 	// Try to store peerset file for all known peers whatsoever
 	// if we got ready (otherwise, don't overwrite anything)
 	if c.readyB {
-		c.peerManager.SavePeerstoreForPeers(c.host.Peerstore().Peers())
+		err := c.peerManager.SavePeerstoreForPeers(c.host.Peerstore().Peers())
+		if err != nil {
+			logger.Warningf("could not save peers to peerstore: %s", err)
```

Comment:
double-logging, since you are logging a similar message already in the other place. leave unchanged, maybe with a comment saying we are ignoring errors because it's a best-effott.